### PR TITLE
Add build job for devtools on quartz

### LINKED ISSUE #145: Fix the devtools build and add a test
After the last uberenv update, we can no longer build the devtools via spack. We should fix this and add a test to make sure this doesn't happen again.

### PR DESCRIPTION
**Pull request by joshessman-llnl**
Resolves #145

### CODE REVIEW

**Comment by joshessman-llnl** [comment]
LGTM

**Comment by joshessman-llnl** [comment]
LGTM

**Comment by joshessman-llnl** [comment]
Devtools build successful: https://lc.llnl.gov/gitlab/smith/serac/pipelines/16740

File: scripts/uberenv/spack_configs/toss_3_x86_64_ib/devtools/packages.yaml [hunk 1]
```diff
@@ -92,6 +92,15 @@ packages:
...
       tar: /usr
    buildable: False
+  python:
+    paths:
+      python@3.8.2: /usr/tce/packages/python/python-3.8.2
+    buildable: False
+  py-sphinx:
+    version: [2.2.0]
+    paths:
+      py-sphinx@2.2.0: /usr/tce/packages/python/python-3.8.2/
```
Comment (jamiebramwell) [inline scripts/uberenv/spack_configs/toss_3_x86_64_ib/devtools/packages.yaml:102]: 👍

**Review by jamiebramwell — APPROVED** [review]
If I'm reading this right, we'll be checking our devtools build after every PR merge? Sounds good to me!

File: scripts/uberenv/spack_configs/toss_3_x86_64_ib/devtools/packages.yaml (hunk 1 above)
Comment (joshessman-llnl) [reply scripts/uberenv/spack_configs/toss_3_x86_64_ib/devtools/packages.yaml:102]: Right, it'll add ~25m to the CI time but the increased frequency will hopefully prevent silent regressions.

Comment (jamiebramwell) [reply scripts/uberenv/spack_configs/toss_3_x86_64_ib/devtools/packages.yaml:102]: If we're on a debug partition, that may take us past the one hour time limit. Are we running on batch?

Comment (joshessman-llnl) [reply scripts/uberenv/spack_configs/toss_3_x86_64_ib/devtools/packages.yaml:102]: I believe so, it's a separate allocation for each stage

**Review by white238 — APPROVED** [review]

**Pull request merged**